server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -72,19 +72,36 @@ private static String toLoggerName(final Class<?> cls) {
     }
 
     /**
-     * Logs a message at the {@link #DEPRECATION} level. The message is also sent to the header warning logger,
+     * Logs a message at the {@link Level#WARN} level. The message is also sent to the header warning logger,
      * so that it can be returned to the client.
      */
     public DeprecationLogger deprecate(
         final DeprecationCategory category,
         final String key,
         final String msg,
-        final Object... params
-    ) {
+        final Object... params) {
+        return logDeprecation(CRITICAL, category, key, msg, params);
+    }
+
+    /**
+     * Logs a message at the {@link Level#INFO} level for less critical deprecations
+     * that likely won't break in next version.
+     * The message is also sent to the header warning logger,
+     * so that it can be returned to the client.
+     */
+    public DeprecationLogger deprecateAtWarnLevel(
```

Comment:
hmm maybe `deprecateNonCritical` is better?

Reply:
Hmm...I wonder about perhaps just using a log4j style e.g.:

```java
deprecationLogger.critical(DeprecationCategory.OTHER, "broken in next version");
deprecationLogger.warn(DeprecationCategory.OTHER, "only somewhat broken in next version");

Reply:
I like the idea overall, but I wonder if this would not make it a bit too clear for developers to know which method to use?
I feel like people are used to `deprecationLogger.deprecate` which helps them to forget about log4j level detail.
Any thoughts on this?

Reply:
I actually like the idea of developers being required to think about their deprecation messages more carefully, because these messages will be exposed far more directly to customers in the future. If we used `critical()` and `warn()`, we could circulate an email about `es@` about it, and maybe update the contributing guide.